activestorage/lib/active_storage/previewer/video_previewer.rb

```diff
@@ -28,21 +28,7 @@ def preview(**options)
 
     private
       def draw_relevant_frame_from(file, &block)
-        ffmpeg_args = [
-          "-i", file.path,
-          "-vf",
-            # Select the first video frame, plus keyframes and frames
-            # that meet the scene change threshold.
-            'select=eq(n\,0)+eq(key\,1)+gt(scene\,0.015),' +
-            # Loop the first 1-2 selected frames in case we were only
-            # able to select 1 frame, then drop the first looped frame.
-            # This lets us use the first video frame as a fallback.
-            "loop=loop=-1:size=2,trim=start_frame=1",
-          "-frames:v", "1",
-          "-f", "image2", "-",
-        ]
-
-        draw self.class.ffmpeg_path, *ffmpeg_args, &block
+        draw self.class.ffmpeg_path, "-i", file.path, *ActiveStorage.video_preview_arguments, &block
```

Comment:
I think separating `"-i", file.path` from `config.active_storage.video_preview_arguments` was a good call. :+1:  What do you think about also separating the trailing `"-"` (which indicates "write to stdout", which, I believe, Active Storage requires)?

Also, for usability, what if `config.active_storage.video_preview_arguments` was a String instead of an Array?  We could then use `Shellwords` to split correctly:

```suggestion
        draw self.class.ffmpeg_path, "-i", file.path, *Shellwords.split(ActiveStorage.video_preview_arguments), "-", &block
```

Reply:
Good idea about extracting the trailing `"-"`.

RE `Shellwords.split()` I've been bitten in the past with this in `dragonfly` (Related: https://zxsecurity.co.nz/research/argunment-injection-ruby-dragonfly/). I guess I just have a bit of PTSD around it! :) Arrays feel more explicit, but of course in this case the string is coming from the developer, and there's no way to provide user-supplied strings here. And `file.path` I assume is trusted from elsewhere. Anyway, it's probably ok :)

Is it also slightly redundant to split the string anyway since it's not having anything interpolated? We could just pass the string straight in.

Reply:
👍🏻 for making it a string and removing the file and stdin arguments.

Reply:
Sounds good to me.

Reply:
I'm tempted to extract `-f image2` also since this just specifies that we want an image out. I guess there might be a scenario where someone doesn't want that though?